src/psyclone/dynamo0p3.py

```diff
@@ -6193,31 +7273,17 @@ def field_bcs_kernel(self, function_space):
                 "Expected a gh_field from which to look-up boundary dofs "
                 "for kernel {0} but got {1}".format(self._kern.name,
                                                     farg.type))
-        new_parent, position = parent.start_parent_loop()
-        new_parent.add(AssignGen(new_parent, pointer=True,
-                                 lhs="boundary_dofs",
-                                 rhs=farg.proxy_name +
-                                 "%vspace%get_boundary_dofs()"),
-                       position=["before", position])
+        self._arglist.append("boundary_dofs_"+farg.name)
 
     def operator_bcs_kernel(self, function_space):
-        ''' Supply necessary additional arguments for the kernel that
-        applies boundary conditions to a LMA operator '''
-        from psyclone.f2pygen import DeclGen, AssignGen
+        '''
+        Supply necessary additional arguments for the kernel that
+        applies boundary conditions to a LMA operator.
+        '''
         # This kernel has only a single LMA operator as argument.
         # Checks for this are performed in ArgOrdering.generate()
         op_arg = self._kern.arguments.args[0]
-        self._arglist.append("boundary_dofs")
-        parent = self._parent
-        parent.add(DeclGen(parent, datatype="integer",
-                           pointer=True, entity_decls=[
-                               "boundary_dofs(:,:) => null()"]))
-        new_parent, position = parent.start_parent_loop()
-        new_parent.add(AssignGen(new_parent, pointer=True,
-                                 lhs="boundary_dofs",
-                                 rhs=op_arg.proxy_name +
-                                 "%fs_to%get_boundary_dofs()"),
-                       position=["before", position])
+        self._arglist.append("boundary_dofs_"+op_arg.name)
```

Comment:
is there a namespace manager requirement here?

Reply:
Probably :-) Changed it to use the namespace manager.